luda-editor/client/src/pages/sequence_edit_page/components/image_picker/mod.rs

```diff
@@ -0,0 +1,138 @@
+use crate::{
+    color,
+    pages::sequence_edit_page::atom::{IMAGES_ATOM, SEQUENCE_ATOM},
+    storage::get_project_image_url,
+};
+use namui::prelude::*;
+use namui_prebuilt::{table::hooks::TableCell, *};
+use rpc::data::{Cut, ImageWithLabels};
+
+const OUTER_PADDING: Px = px(8.0);
+const INNER_PADDING: Px = px(4.0);
+const THUMBNAIL_WH: Wh<Px> = Wh {
+    width: px(144.0),
+    height: px(144.0),
+};
+
+#[namui::component]
+pub struct ImagePicker<'a> {
+    pub wh: Wh<Px>,
+    pub project_id: Uuid,
+    pub cut: &'a Cut,
+    pub on_event: &'a dyn Fn(Event),
+}
+
+pub enum Event {
+    Close,
+}
+pub enum InternalEvent {
+    ImageThumbnailClicked { image_id: Uuid },
+}
+
+impl Component for ImagePicker<'_> {
+    fn render(self, ctx: &RenderCtx) -> RenderDone {
+        let Self {
+            wh,
+            project_id,
+            cut,
+            on_event,
+        } = self;
+        let (images, _set_images) = ctx.atom(&IMAGES_ATOM);
+
+        let on_internal_event = |event: InternalEvent| match event {
+            InternalEvent::ImageThumbnailClicked { image_id } => {
+                add_image_to_cut(image_id, cut.id);
+                on_event(Event::Close);
+            }
+        };
+
+        let max_items_per_row = ctx.memo(|| (wh.width / (THUMBNAIL_WH.width)).floor() as usize);
```

Comment:
This must be trouble because this memo doesn't have any signal so it won't update when wh is changed.

Reply:
Ah, true